src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
 public class SalaryInfo {
+    private static final String FORMATTER = "dd.MM.yyyy";
+    private static final int DATA = 0;
+    private static final int NAME = 1;
+    private static final int HOURS = 2;
+    private static final int price = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SimpleDateFormat format = new SimpleDateFormat(FORMATTER);
+        int[] salary = new int[names.length];
+        for (int i = 0; i < data.length; i++) {
+            String[] dataLine = data[i].split(" ");
+            boolean correctDate;
+            try {
+                correctDate = format.parse(dateFrom).compareTo(format.parse(dataLine[DATA])) <= 0
```

Comment:
`format.parse(dateFrom)` and `format.parse(dateTo)` - we can do this only once in the very beginning of the method